Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1

```diff
@@ -73,10 +76,59 @@ function Write-WarningLog {
         [array] $Arguments
     )
 
-    if ($null -eq $Arguments) {
-        Write-Warning -Message $message
+    if ($null -ne $Arguments) {
+        $Message = [string]::Format($Message, $Arguments)
+    }
+    
+    Write-Warning -Message $Message
+
+    if ($WarningPreference -ne 'Continue') {
+        Write-LogToFile -Message $Message -LogType 'Warning'
     }
-    else {
-        Write-Warning -Message ([string]::Format($Message, $Arguments))
+}
+
+<#
+.SYNOPSIS
+Saves the messages to a log file
+
+.DESCRIPTION
+Saves the messages to a log file, because Invoke-DscResource runs this in a separate runspace
+where the streams are not accessable.
+
+.PARAMETER Message
+Message to save into the log file.
+
+.PARAMETER LogType
+Type of the stream: Verbose, Warning etc..
+#>
+function Write-LogToFile {
+    Param(
+        [string]
+        $Message,
+
+        [string]
+        $LogType
+    )
+
+    # do nothing in case no temp folder is available
+    if ($null -eq $env:TEMP) {
+        return
     }
+
+    $logFilePath = Join-Path -Path $env:TEMP -ChildPath "__VMware.vSphereDSC_$LogType.txt"
```

Comment:
It's a good practice to add timestamp or GUID (even better) to temp file names to avoid collisions. 

https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename?view=net-5.0 is the easiest and safest way though.

Reply:
The problem here is that I can't share variable between the main and the secondary runspaces.  If I use a guid or timestamp on the name I won't be able to find out what it is during the main execution. That is why the names must be static.